chore(sdk-core): refactor BaseCoin from bitgo to sdk-core.

ducnguyen-bitgo opened the pull request:
- Moved baseCoin to sdk-core.
- Moved enterprise and enterprises to sdk-core.
- Moved internal and keycard to sdk-core.
- Moved keychains to sdk-core.
- Moved markets to sdk-core.
- Moved pendingApproval and pendingApprovals to sdk-core.
- Moved affirmation and affirmations to sdk-core.
- Moved settlement and settlements to sdk-core.
- Moved tradingAccount to sdk-core.
- Moved tradingPartner, and tradingPartners to sdk-core.
- Moved blsUtils, mpcUtils, opengpgUtils, promise-utils to sdk-core.
- Moved tssUtils, util to sdk-core.
- Moved wallet, and wallets to sdk-core.
- Moved webhooks to sdk-core.
- Moved bip32util, bitcoin, ecdh, and tss to sdk-core.

Ticket: BG-48167

ducnguyen-bitgo committed:
chore(sdk-core): refactor BaseCoin from bitgo to sdk-core

- Moved baseCoin to sdk-core.
- Moved enterprise and enterprises to sdk-core.
- Moved internal and keycard to sdk-core.
- Moved keychains to sdk-core.
- Moved markets to sdk-core.
- Moved pendingApproval and pendingApprovals to sdk-core.
- Moved affirmation and affirmations to sdk-core.
- Moved settlement and settlements to sdk-core.
- Moved tradingAccount to sdk-core.
- Moved tradingPartner, and tradingPartners to sdk-core.
- Moved blsUtils, mpcUtils, opengpgUtils, promise-utils to sdk-core.
- Moved tssUtils, util to sdk-core.
- Moved wallet, and wallets to sdk-core.
- Moved webhooks to sdk-core.
- Moved bip32util, bitcoin, ecdh, and tss to sdk-core.

Ticket: BG-48167

ducnguyen-bitgo committed:
chore(sdk-core): re-export request tracer for backward compat

- Re-export RequestTracer in bitgo for backward compat.
- Fixed the barrel imports in BaseCoin.

Ticket: BG-48167

bitgoAaron commented on modules/sdk-core/src/bitgo/utils/abstractUtxoCoinUtil.ts:
```diff
@@ -0,0 +1,7 @@
+import * as utxolib from '@bitgo/utxo-lib';
+import ScriptType2Of3 = utxolib.bitgo.outputScripts.ScriptType2Of3;
+import { isChainCode, scriptTypeForChain } from '@bitgo/utxo-lib/dist/src/bitgo';
```
let's avoid importing from `/dist` folders. Can we instead make sure utxo-lib exports these types correctly?

ducnguyen-bitgo replied:
Those functions are exported under `utxo-lib/src/bitgo` so I guess we could just do `import { bitgo } from '@bitgo/utxo-lib'` and then destructure those functions from `bitgo`?

bitgoAaron commented on modules/sdk-core/package.json:
```diff
@@ -37,15 +37,21 @@
     "bs58": "^4.0.1",
     "create-hmac": "^1.1.7",
+    "debug": "^3.1.0",
+    "libsodium-wrappers-sumo": "^0.7.9",
```
same here. Remove this from bitgo's package.json

ducnguyen-bitgo replied:
I think `libsodium-wrappers-sumo` is from account-lib's package.json. I'm removing it now.

bitgoAaron commented:
looks like all of the files under `sdk-core/src/account-lib/mpc` were manually copied instead of `git mv`, thus losing the history. Can you redo these as well? We need to keep track of as much as possible

bitgoAaron reviewed (changes requested):
see comments

ducnguyen-bitgo committed:
chore(sdk-core): remove mpc from sdk-core and unused packages

- Removed mpc from sdk-core.
- Removed unused packages from bitgo and account-lib.

Ticket: BG-48167

ducnguyen-bitgo committed:
chore(sdk-core): git mv mpc from account-lib to sdk-core

- Merged branch master into ducnguyen/refactor-basecoin-to-sdk-core.
- Moved mpc from account-lib to sdk-core.
- Fixed import issues in unit tests.

Ticket: BG-48167

ducnguyen-bitgo committed:
Merge branch 'master' into ducnguyen/refactor-basecoin-to-sdk-core

ducnguyen-bitgo committed:
chore(sdk-core): remove mpc/tss from sdk-core

- Removed mpc/tss from sdk-core in order to use git mv.

Ticket: BG-48167

ducnguyen-bitgo committed:
chore(sdk-core): git mv mpc/tss to sdk-core

- Moved mpc/tss from account-lib to sdk-core.

Ticket: BG-48167

ducnguyen-bitgo committed:
chore(sdk-core): fix import issues with account-lib mpc

- Exported KeyShare from mpc instead of the one from iKeyChains.
- Fixed imports in bitgo unit tests.

Ticket: BG-48167

bitgoAaron commented on modules/sdk-core/src/api/bip32path.ts:
```diff
@@ -0,0 +1,42 @@
+/**
```
the file should be `git mv`'d, not copy/pasted.

bitgoAaron reviewed (changes requested):
2 more copy/pasted files need to be git mv'd to preserve history.

ducnguyen-bitgo committed:
chore(sdk-core): remove bip32path in sdk-core api to handle git mv

- Removed bip32path in sdk-core api.

Ticket: BG-48167

ducnguyen-bitgo committed:
chore(sdk-core): git mv bip32path to sdk-core api

- Git mv bip32path from sdk-api to sdk-core api.
- Fixed import issues in bitgo.

Ticket: BG-48167

ducnguyen-bitgo commented on modules/sdk-core/src/bitgo/utils/util.ts:
```diff
@@ -0,0 +1,199 @@
+/**
```
@bitgoAaron The reason we didn't git-mv it was to keep it backward compatible for bitgo-express until the next release of sdk-core.

ducnguyen-bitgo committed:
chore: merge master into ducnguyen/refactor-basecoin-to-sdk-core

- Merged branch master.

Ticket: BG-48167

ducnguyen-bitgo committed:
chore(sdk-core): fix baseCoin import issues

- Fixed baseCoin import issues in fiateur and tfiateur.

Ticket: BG-48167
